### PR TITLE
refactor(i18n)!: reduce list of supported numbering systems

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #8216

## Summary

Reduce the list of supported numbering systems to `arab`, `arabext`, and `latn` after a discussion in the Internationalization Teams channel.

BREAKING CHANGE: Removed support for the following numbering systems: `bali`, `beng`, `deva`, `fullwide`, `gujr`, `guru`, `hanidec`, `khmr`, `knda`, `laoo`, `limb`, `mlym`, `mong`, `mymr`, `orya`, `tamldec`, `telu`, `thai`, `tibt`

### CODE REVIEW

File: packages/calcite-components/src/components/input-time-picker/input-time-picker.e2e.ts [hunk 1]
```diff
@@ -665,11 +665,10 @@ describe("calcite-input-time-picker", () => {
     expect(await getInputValue(page)).toBe("٠٢:٣٠:٢٥ م");
 
     inputTimePicker.setProperty("lang", "zh-HK");
-    inputTimePicker.setProperty("numberingSystem", "hanidec");
```
Comment (benelan) [inline packages/calcite-components/src/components/input-time-picker/input-time-picker.e2e.ts:668]: I think this needs to be

```ts
inputTimePicker.setProperty("numberingSystem", "latn");
```

to match the expectation

Comment (jcfranco) [reply packages/calcite-components/src/components/input-time-picker/input-time-picker.e2e.ts:668]: Good catch. Fixing!

**Review by jcfranco — APPROVED** [review]
✂️#️⃣

**Comment by jcfranco** [comment]
For posterity: 9270bb60bc7127a89e2a97825433aa4d4d1fa511 is the commit to cherry pick In case we need to restore `thai`.

**Pull request merged**